dotnet-build.sh

```diff
@@ -65,12 +91,19 @@ echo "building $CS_PROJECT_FILE"
 dotnet build -c $RELEASE --no-restore $CS_PROJECT_FILE
 
 # run tests
-echo "running tests"
-for f in *.Test/*.csproj
-do
-  echo "Processing $f file..."
-  dotnet test "$f" --no-restore -c $RELEASE -r "$DIST/test-results"  
-done
+if [ $WITH_TESTING ]
+then
+  echo "running tests"
+  for f in *.Test/*.csproj
+  do
+    echo "Processing $f file..."
+    dotnet test "$f" -c $RELEASE -r "$DIST/test-results" --filter "TestCategory!=[Integration]" /p:CollectCoverage=true /p:CoverletOutputFormat="opencover" /p:CoverletOutput="$DIST/test-coverlet"
+  done
+    
+  # generate coverage report
+  dotnet tool install dotnet-reportgenerator-globaltool --tool-path "$DIST/tools"
+  $DIST/tools/reportgenerator -reports:"**/*.opencover*.xml" -targetdir:dist/coverage -reporttypes:"Cobertura;HTMLInline;HTMLChart;SonarQube"
```

Comment:
test results have to be stored in a specific location for bitbucket to pick them up. This changes the output-dir.